includedBuild/gradleConfiguration/src/main/kotlin/com/badoo/reaktive/configuration/MppConfigurationPlugin.kt

```diff
@@ -39,6 +40,21 @@ class MppConfigurationPlugin : Plugin<Project> {
             setupJsTarget(project)
             setupLinuxX64Target(project)
 
+            if (Target.shouldDefineTarget(project, Target.JVM)) {
+                project.tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
+                    kotlinOptions {
+                        /*
+                         * Starting from Kotlin 1.5.0 the default jvmTarget is 1.8, and the 1.6 target is deprecated.
+                         * For some reason some JVM tests are failing with
+                         * java.lang.NoSuchMethodError: 'int java.lang.Integer.compareTo(int)'.
+                         * The Integer.compareTo method was added in Java 1.8.
+                         * We should find a fix, or avoid using the Integer.compareTo method.
+                         */
+                        jvmTarget = "1.6"
```

Comment:
I did not get this part.
They are failing because of Java 1.8?

Reply:
I'm not sure why does it fail. I suspect that tests are somehow compiled with 1.6 and so the method is not found. Also not sure why it does not fail in master. Could not google anything. Will appreciate any help here.

Reply:
```kotlin
private val comparator = Comparator<Int> { a, b ->  a.compareTo(b) }
```

It fixes the issue. I do not know why method reference does not work, might be a new IR backend bug related to method referencing.

Reply:
Interesting, I did not even try since the error message is pretty much descriptive. Will apply, thanks!

Reply:
Filed [KT-46699](https://youtrack.jetbrains.com/issue/KT-46699)